src/core/Tokenizer.ts

```diff
@@ -10,83 +10,91 @@ const toCanonicalKeyword = (text: string) => equalizeWhitespace(text.toUpperCase
 
 /** Struct that defines how a SQL language can be broken into tokens */
 interface TokenizerOptions {
-  reservedKeywords: string[];
+  // Main clauses that start new block, like: SELECT, FROM, WHERE, ORDER BY
   reservedCommands: string[];
+  // Logical operator keywords, defaults to: [AND, OR]
   reservedLogicalOperators?: string[];
+  // Keywords in CASE expressions that begin new line, like: WHEN, ELSE
   reservedDependentClauses: string[];
+  // Keywords that create newline but no indentaion of their body.
+  // These contain set operations like UNION and various joins like LEFT OUTER JOIN
   reservedBinaryCommands: string[];
+  // keywords used for JOIN conditions, defaults to: [ON, USING]
   reservedJoinConditions?: string[];
-  stringTypes: regexFactory.StringPatternType[];
+  // all other reserved words (not included to any of the above lists)
+  reservedKeywords: string[];
+  // Types of quotes to use for strings
+  stringTypes: regexFactory.QuoteType[];
+  // Types of quotes to use for quoted identifiers
+  identifierTypes: regexFactory.QuoteType[];
+  // Open-parenthesis characters, like: (, [, {
   blockStart?: string[];
+  // Close-parenthesis characters, like: ), ], }
   blockEnd?: string[];
-  indexedPlaceholderTypes?: string[];
-  namedPlaceholderTypes?: string[];
+  // True to allow for positional "?" parameter placeholders
+  positionalPlaceholders?: boolean;
+  // Prefixes for numbered parameter placeholders to support, e.g. :1, :2, :3
+  numberedPlaceholderTypes?: ('?' | ':' | '$')[];
+  // Prefixes for named parameter placeholders to support, e.g. :name
+  namedPlaceholderTypes?: (':' | '@' | '$')[];
+  // Prefixes for quoted parameter placeholders to support, e.g. :"name"
+  // The type of quotes will depend on `identifierTypes` option.
+  quotedPlaceholderTypes?: (':' | '@' | '$')[];
+  // Line comment types to support, defaults to --
   lineCommentTypes?: string[];
-  specialWordChars?: { prefix?: string; any?: string; suffix?: string };
+  // Additional characters to support in identifiers
+  specialIdentChars?: regexFactory.IdentChars;
+  // Additional multi-character operators to support, in addition to <=, >=, <>, !=
   operators?: string[];
+  // Allows custom modifications on the token array.
+  // Called after the whole input string has been split into tokens.
+  // The result of this will be the output of the tokenizer.
   preprocess?: (tokens: Token[]) => Token[];
 }
 
+type PlaceholderPattern = { regex: RegExp; parseKey: (s: string) => string };
```

Comment:
convert to interface

Reply:
Yeah... probably better for consistency.